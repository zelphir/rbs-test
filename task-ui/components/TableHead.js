import React from 'react'
import PropTypes from 'prop-types'

const TableHeader = ({ headers }) => (
  <thead>
    <tr>{Object.keys(headers).map(header => <th key={header}>{header}</th>)}</tr>
  </thead>
)

TableHeader.propTypes = {
  headers: PropTypes.shape({
    id: PropTypes.number.isRequired,
    amount: PropTypes.number.isRequired,
    date: PropTypes.string.isRequired,
    business: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
    account: PropTypes.string.isRequired
  }).isRequired
}

export default TableHeader
