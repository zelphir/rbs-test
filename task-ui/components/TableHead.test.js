import React from 'react'
import { shallow } from 'enzyme'
import TableHead from './TableHead'

const headers = {
  id: 1,
  amount: 85.91,
  date: '2012-02-02T00:00:00.000Z',
  business: 'Bode - Kreiger',
  name: 'Savings Account 6670',
  type: 'withdrawal',
  account: '86389677'
}

describe('<TableHead />', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<TableHead headers={headers} />)
    expect(wrapper).toMatchSnapshot()
  })
})
