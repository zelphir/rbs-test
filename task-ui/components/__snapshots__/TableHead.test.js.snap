// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableHead /> should match the snapshot 1`] = `
<thead>
  <tr>
    <th
      key="id"
    >
      id
    </th>
    <th
      key="amount"
    >
      amount
    </th>
    <th
      key="date"
    >
      date
    </th>
    <th
      key="business"
    >
      business
    </th>
    <th
      key="name"
    >
      name
    </th>
    <th
      key="type"
    >
      type
    </th>
    <th
      key="account"
    >
      account
    </th>
  </tr>
</thead>
`;
